### PR TITLE
RELENG-4590 redhat registration

### PR DESCRIPTION
**Pull request by bertranddemiddelaer**

### CODE REVIEW

**Review by tcarmet — COMMENTED** [review]
the idea looks more than good to me. 

I would suggest adding RED_HAT username and password on the conf of the service and adding in the script

File: srcs/runners_manager/vm_creation/openstack.py [hunk 1]
```diff
@@ -19,7 +19,8 @@ class OpenstackManager(object):
     nova_client: novaclient.client.Client
     neutron: neutronclient.v2_0.client.Client
```
Comment (erwan-b) [inline srcs/runners_manager/vm_creation/openstack.py:20]: add `redhat_username` and `redhat_password` here

File: srcs/runners_manager/vm_creation/openstack.py [hunk 2]
```diff
@@ -41,12 +42,15 @@ def __init__(self, project_name, token, username, password, region):
                     project_domain_id='default')
             )
 
+        self.redhat_username = redhat_username
+        self.redhat_password = redhat_password
         self.nova_client = novaclient.client.Client(version=2, session=session, region_name=region)
         self.neutron = neutronclient.v2_0.client.Client(session=session, region_name=region)
 
     @staticmethod
     def script_init_runner(runner: Runner, token: int,
```
Comment (erwan-b) [inline srcs/runners_manager/vm_creation/openstack.py:51]: Maybe remove the `@statimethod`  ?

File: templates/init_runner_script.sh [hunk 3]
```diff
@@ -21,12 +23,49 @@ then
...
+
+[Install]
+WantedBy=multi-user.target
+EOF'
+sudo systemctl daemon-reload
+sudo systemctl enable redhat_registration.service
+sudo systemctl start redhat_registration.service
+if [ "${LINUX_OS_VERSION}" = "7" ]
+then
+subscription-manager repos --enable=rhel-7-server-extras-rpms --enable=rhel-7-server-optional-rpms
```
Comment (tcarmet) [inline templates/init_runner_script.sh:50]:
```suggestion
# Enable repo to install docker
subscription-manager repos --enable=rhel-7-server-extras-rpms --enable=rhel-7-server-optional-rpms
```

**Comment by bertranddemiddelaer** [comment]
> * Add redhat config on config_samples
> * Remove static_method
> 
> And then LGTM, already approving

Done.

**Review by erwan-b — APPROVED** [review]

**Pull request merged**